frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/FeatureEnvironmentVariants.tsx

```diff
@@ -46,14 +54,29 @@ export const FeatureEnvironmentVariants = () => {
     );
     const { patchFeatureEnvironmentVariants, overrideVariantsInEnvironments } =
         useFeatureApi();
+    const { refetch: refetchChangeRequests } =
+        usePendingChangeRequests(projectId);
+    const { addChange } = useChangeRequestApi();
+    const { isChangeRequestConfigured } = useChangeRequestsEnabled(projectId);
 
     const [searchValue, setSearchValue] = useState('');
     const [selectedEnvironment, setSelectedEnvironment] =
-        useState<IFeatureEnvironment>();
+        useState<IFeatureEnvironmentWithCrEnabled>();
     const [selectedVariant, setSelectedVariant] = useState<IFeatureVariant>();
     const [modalOpen, setModalOpen] = useState(false);
     const [deleteOpen, setDeleteOpen] = useState(false);
 
+    const environments: IFeatureEnvironmentWithCrEnabled[] = useMemo(
+        () =>
+            feature?.environments?.map(environment => ({
+                ...environment,
+                crEnabled:
+                    uiConfig.flags.crOnVariants &&
+                    isChangeRequestConfigured(environment.name),
+            })) || [],
+        [feature.environments, uiConfig.flags.crOnVariants]
+    );
```

Comment:
Oh, this is where the magic happens decorating the feature with `crEnabled` and this basically includes the `isChangeRequestConfigured` in env and feature toggle is enabled

Reply:
Yeah exactly. This way we can clean this up once we make the flag GA and the logic is the same. Does this answer https://github.com/Unleash/unleash/pull/2989#discussion_r1086617295?

Reply:
yes